### PR TITLE
Ignore pubsub timeout errors

### PR DESCRIPTION
**Pull request by ichinaski**
Handles DEADLINE_EXCEEDED rendezvouz exception.

Follow up on #19

### CODE REVIEW

File: psq/worker.py [hunk 1]
```diff
@@ -42,7 +45,12 @@ def _safe_dequeue(self):
             wait_exponential_multiplier=1000, wait_exponential_max=10000,
             retry_on_exception=lambda e: not isinstance(e, KeyboardInterrupt))
         def inner():
-            return self.queue.dequeue(max=self.tasks_per_poll, block=True)
+            try:
+                return self.queue.dequeue(max=self.tasks_per_poll, block=True)
+            except errors.GaxError as e:
```
Comment (jonparrott) [inline psq/worker.py]: @dhermes does `google-cloud` not wrap this exception?

Comment (dhermes) [inline psq/worker.py]: I don't have much detail on what is causing this exception, but I'll take a stab in the dark.

I'd guess you're talking about the DEADLINE_EXCEED on pull? See the ongoing discussion: https://github.com/GoogleCloudPlatform/google-cloud-python/pull/2576#issuecomment-262029203

Comment (ichinaski) [inline psq/worker.py]: @dhermes indeed, it seems like the same issue I'm facing. 

I'm not sure whether this is expected and `psq` should be in charge of handling the exception or, as you mention in that thread, just return an empty set from `pubsub`. The latter will remove the necessity of handling the exception in client libraries.

Comment (dhermes) [inline psq/worker.py]: Yeah we are going to just swallow the `DEADLINE_EXCEEDED`. In some sense, that is the appropiate response and the JSON-over-HTTP API is the one getting it wrong on a timeout.

Comment (ichinaski) [inline psq/worker.py]: Awesome, closing this PR then. Thanks for your help!

**Comment by ichinaski** [comment]
This issue will be addressed in the `pubsub` library: https://github.com/GoogleCloudPlatform/google-cloud-python/pull/2576#issuecomment-262029203

File: psq/worker.py (hunk 1 above)
Comment (jonparrott) [inline psq/worker.py]: Awesome, thanks everyone. :D

**Pull request closed without merging**